app/components/toolbarSearch/toolbarSearchDirective.js

```diff
@@ -36,34 +54,77 @@
                 /**
                  *
                  */
-                vm.onKeyPress = function($event) {
+                function onKeyPress($event) {
                     // only fire the search if Enter-key (13) is pressed
                     if (getKeyboardCodeFromEvent($event) === 13) {
                         vm.doSearch();
+                    }else if (getKeyboardCodeFromEvent($event) === 8) {
+                        removeKeyWordFromDeleteKey();
                     }
-                };
+                }
 
                 /**
                  *
                  */
-                vm.doSearch = function() {
-                    // if no input is given
-                    // if (vm.searchInput.length === 0) {
-                    //    return false;
-                    // }
+                function doSearch() {
+                    var fiterText;
+                    if (vm.textSearchInput.value.length) {
+                        fiterText = vm.filter.text || '';
+                        vm.filter.text = fiterText + ' "' + vm.textSearchInput.value + '"';
+                        vm.textSearchInput = {value: '', previousLength: 0};
+                    }
                     HeatMapSourceGenerator.search(vm.filter.text);
-                };
+                }
 
-                vm.reset = function() {
+                function reset() {
                     searchFilter.resetFilter();
                     HeatMapSourceGenerator.search();
                     // Reset the map
                     MapService.resetMap();
-                };
+                }
 
-                vm.showtoolbarSearchInfo = function() {
+                function showtoolbarSearchInfo() {
                     InfoService.showInfoPopup('textsearch');
-                };
+                }
+
+                function keyWordStringToArray(keyWordString) {
+                    keyWordString = keyWordString || '';
+                    return keyWordString.split('"').filter(function(val){
+                        return val !== '' && val !== ' ';
+                    });
+                }
+
+                function removeKeyWord(keyword) {
```

Comment:
If the list of keywords in the search is ['test1', 'test2', 'test3'], and we want delete 'test2', we pass this value and the and search in HeatMapSourceGenerator `'"test1" "test3"'`